src/lib/card.js

```diff
@@ -0,0 +1,191 @@
+const initCanvas = (canvas, size, ratio) => {
+  const { x, y } = size;
+  // let ctx = canvas.getContext('2d')
+
+  const dpi200 = window.devicePixelRatio * (200 / 72);
+
+  canvas.width = x * dpi200;
+  canvas.height = y * dpi200;
+  canvas.style.width = x + 'px';
+  canvas.style.height = y + 'px';
+
+  canvas.getContext('2d').scale(dpi200, dpi200);
+
+  return canvas;
+};
+
+// takes a
```

Comment:
This doesn't seem finished. Maybe we want comments for most of these functions/data structures as well. Comments in walk and rand are good!

Reply:
Resolved in 4543574